udata/core/dataset/views.py

```diff
@@ -86,8 +88,86 @@ def get_context(self):
         context['can_edit'] = DatasetEditPermission(self.dataset)
         context['can_edit_resource'] = ResourceEditPermission
         context['discussions'] = Discussion.objects(subject=self.dataset)
+
         return context
 
+    def get_json_ld(self):
+        dataset = self.dataset
+
+        result = {"@context": "http://schema.org",
+                  "@type": "Dataset",
+                  "@id": str(dataset.id),
+                  "description": dataset.description,
+                  "alternateName": dataset.slug,
+                  "dateCreated": dataset.created_at.isoformat(),
+                  "dateModified": dataset.last_modified.isoformat(),
+                  "url": url_for('datasets.show', dataset=dataset, _external=True),
+                  "name": dataset.title,
+                  "keywords": ', '.join(dataset.tags),
+                  "distribution": map(self.get_json_ld_resource, dataset.resources),
+                  # This value is not standard
+                  "extras": map(self.get_json_ld_extra, dataset.extras.items())
+        }
+
+        if dataset.license and dataset.license.url:
+            result["license"] = dataset.license.url
+
+        if dataset.organization:
+            view = OrganizationDetailView()
+            author = view.get_json_ld(dataset.organization)
+        elif dataset.owner:
+            view = UserActivityView()
+            author = view.get_json_ld(dataset.owner)
+        else:
+            author = None
+
+        if author:
+            result['author'] = author
+
+        return result
+
+    @staticmethod
+    def get_json_ld_resource(resource):
+
+        result = {
+            "@type": "DataDownload",
+            "@id": str(resource.id),
+            "url": resource.url,
+            "name": resource.title or _('Nameless resource'),
+            "contentUrl": resource.url,
+            "encodingFormat": resource.format or '',
```

Comment:
I think the keys should only be present if not empty (I comment it, but it's valid for all empty values)